stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/MediaGalleryPreviewActivity.kt

```diff
@@ -0,0 +1,1413 @@
+                                        }
+
+                                        if (currentScale == DefaultZoomScale) {
+                                            translation = Offset(0f, 0f)
+                                        }
+                                    }
+                                }
+                            }
+                        },
+                    painter = painter,
+                    contentDescription = null
+                )
+
+                Log.d("isCurrentPage", "${page != pagerState.currentPage}")
+
+                if (pagerState.currentPage != page) {
+                    currentScale = DefaultZoomScale
+                    translation = Offset(0f, 0f)
+                }
+            }
+        }
+    }
+
+    /**
+     * Represents an individual page containing video player with media controls.
+     *
+     * @param attachment The video attachment to be played.
+     * @param pagerState The state of the pager that contains this page
+     * @param page The page an instance of this content is located on.
+     * @param onPlaybackError Handler for playback errors.
+     */
+    @Composable
+    private fun VideoPreviewContent(
+        attachment: Attachment,
+        pagerState: PagerState,
+        page: Int,
+        onPlaybackError: () -> Unit,
+    ) {
+        val context = LocalContext.current
+
+        var hasPrepared by remember {
+            mutableStateOf(false)
+        }
+
+        var userHasClickedPlay by remember {
+            mutableStateOf(false)
+        }
+
+        var shouldShowProgressBar by remember {
+            mutableStateOf(false)
+        }
+
+        var shouldShowPreview by remember {
+            mutableStateOf(true)
+        }
+
+        var shouldShowPlayButton by remember {
+            mutableStateOf(true)
+        }
+
+        val mediaController = remember {
+            createMediaController(context)
+        }
+
+        val videoView = remember {
+            VideoView(context)
+        }
+
+        val contentView = remember {
+
+            val frameLayout = FrameLayout(context).apply {
+                layoutParams = FrameLayout.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT
+                )
+            }
+            videoView.apply {
+                setVideoURI(Uri.parse(attachment.assetUrl))
+                this.setMediaController(mediaController)
+                setOnErrorListener { _, _, _ ->
+                    shouldShowProgressBar = false
+                    onPlaybackError()
+                    true
+                }
+                setOnPreparedListener {
+                    // Don't remove the preview unless the user has clicked play previously,
+                    // otherwise the preview will be removed whenever the video has finished downloading.
+                    if (!hasPrepared && userHasClickedPlay) {
```

Comment:
It might seem unnecessary to do these complex checks, but unfortunately preparation is done in the background and the listener can be triggered at any time and remove the preview.

We unfortunately don't have any sort of listener that informs us when playback has started, only when it _can_ be started, i.e. media has been prepared.